particle-iot/particle-usb
# feature/sc-104328/node-usb-v2-line-e2e-tests

## Description

Adds end-to-end test suites for Browser and Node.js usage.


## How to test

1. Reinstall dependencies: `npm run reinstall` (or `npm i` if this is your first time - see [docs](https://github.com/particle-iot/particle-usb#installing))
2. Run tests: `npm run test:ci`
3. Review the [e2e test instructions](https://github.com/particle-iot/particle-usb/blob/feature/sc-104328/node-usb-v2-line-e2e-tests/test/e2e/README.md)
4. Run `e2e` tests _with_ device(s) connected ([docs](https://github.com/particle-iot/particle-usb/blob/feature/sc-104328/node-usb-v2-line-e2e-tests/test/e2e/README.md#running-with-a-device))
5. Run `e2e` tests _without_ device(s) connected ([docs](https://github.com/particle-iot/particle-usb/blob/feature/sc-104328/node-usb-v2-line-e2e-tests/test/e2e/README.md#running-without-a-device))

**Outcome**

All tests should pass, `package-lock.json` should not show changes after reinstalling, all commands should do what you expect, docs are clear


## Related / Discussions

https://app.shortcut.com/particle/story/104328
